support for per mode restrictions

gknisely opened the pull request:
Fixes #747 

Updated lua to process restriction:hgv, restriction:emergency, restriction:taxi, restriction:motorcar, restriction:bus, restriction:bicycle, and restriction:hazmat tags.  In pbfgraphparser.cc, if the above tags are found, they are saved as complex restrictions and the mode is set based on the type in restriction:<type> 

Auto route
![image](https://cloud.githubusercontent.com/assets/2676277/26800699/402987ec-4a08-11e7-8bb6-37830cb9ec9a.png)

Truck route
![image](https://cloud.githubusercontent.com/assets/2676277/26800720/50bbbc38-4a08-11e7-9128-e51782a06b06.png)

dnesbitt61 commented:
:shipit:

mmajan87 commented:
Hi there, thanks for this. However, I tried the following: https://www.openstreetmap.org/directions?engine=mapzen_car&route=42.45411%2C-2.42805%3B42.45371%2C-2.42824#map=18/42.45343/-2.42900

and it looks like it is taking into account a forbidden turn for buses when getting car routing. Can you please verify if the turn restriction in the intersection is not well defined (I used restriction:bus...).

Thanks in advance

gknisely commented:
Hello @mmajan87 
Restrictions are represented only by relations as described here:  http://wiki.openstreetmap.org/wiki/Relation:restriction Therefore, I have removed the restriction from the node (https://www.openstreetmap.org/node/420772858) and updated the relation (https://www.openstreetmap.org/relation/7282887).  Please confirm that it looks correct.  

Please note that this code will not be released until next week and then we must process the data after updating our data creation server.  I will let you know when this is done.  

Thank you for your time!
--Greg
